compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -1732,3 +1735,38 @@ fn opaque_type_cycle_error(tcx: TyCtxt<'_>, def_id: LocalDefId, span: Span) -> E
     }
     err.emit()
 }
+
+pub(super) fn check_generator_obligations(tcx: TyCtxt<'_>, def_id: LocalDefId) {
+    debug_assert!(tcx.sess.opts.unstable_opts.drop_tracking_mir);
+    debug_assert!(matches!(tcx.def_kind(def_id), DefKind::Generator));
+
+    let typeck = tcx.typeck(def_id);
+    let param_env = tcx.param_env(def_id);
+
+    let generator_interior_predicates = &typeck.generator_interior_predicates[&def_id];
+    debug!(?generator_interior_predicates);
+
+    let infcx = tcx
+        .infer_ctxt()
+        // typeck writeback gives us predicates with their regions erased.
+        // As borrowck already has checked lifetimes, we do not need to do it again.
+        .ignoring_regions()
+        // Bubble opaque types to ignore them later.
+        .with_opaque_type_inference(DefiningAnchor::Bubble)
+        .build();
+
+    let mut fulfillment_cx = <dyn TraitEngine<'_>>::new(infcx.tcx);
+    for (predicate, cause) in generator_interior_predicates {
+        let obligation = Obligation::new(cause.clone(), param_env, *predicate);
+        fulfillment_cx.register_predicate_obligation(&infcx, obligation);
+    }
+    let errors = fulfillment_cx.select_all_or_error(&infcx);
+    debug!(?errors);
+    if !errors.is_empty() {
+        infcx.err_ctxt().report_fulfillment_errors(&errors, None, false);
+    }
+
+    // Opaque types may have gotten their hidden types constrained, but we can ignore them
+    // safely as they will get constrained elsewhere, too.
```

Comment:
Can you expand on this? Do we expect that any constraints that occur here will have already occured during the type-checking of the generator body?

Reply:
TBH, I'm not sure here.  Borrowck should have constrained the opaques, but I don't know if we can rule out an incompatible hidden type inference.

Reply:
Are you actually getting any inferences? If not, you can use a different DefiningAnchor scheme

Reply:
I mean... you should probably not use `Bubble`, but `Bind(def_id)`